### PR TITLE
feature(motor-control): add presice brushed linear moves and can messages in physical units

### PR DESCRIPTION
**Pull request by ryanthecoder**
Replaces PR https://github.com/Opentrons/ot3-firmware/pull/443
and provided the additional features of taking commands as physical units instead of directly encoder pulses
RLAB-44
RLAB-64

### CODE REVIEW

File: gripper/firmware/motor_timer_hardware.c [hunk 1]
```diff
@@ -285,10 +277,10 @@ void update_pwm(uint32_t duty_cycle) {
     // (in pwm mode 1, output is high while cnt < ccr, so we don't want
     // to let cnt = ccr if the pwm value is 100%)
     htim1.Instance->CCR1 = clamp(duty_cycle, 0, htim1.Init.Period + 1);
-    htim1.Instance->EGR = TIM_EGR_UG;
+    //htim1.Instance->EGR = TIM_EGR_UG;
```
Comment (sfoster1) [inline gripper/firmware/motor_timer_hardware.c:280]: If we don't need to do this, let's delete the line

File: gripper/firmware/motor_timer_hardware.c [hunk 2]
```diff
@@ -285,10 +277,10 @@ void update_pwm(uint32_t duty_cycle) {
...
 
     htim3.Instance->CCR1 = clamp(duty_cycle, 0, htim3.Init.Period + 1);
-    htim3.Instance->EGR = TIM_EGR_UG;
+    //htim3.Instance->EGR = TIM_EGR_UG;
```
Comment (sfoster1) [inline gripper/firmware/motor_timer_hardware.c:283]: same hting

**Review by sfoster1 — APPROVED** [review]
Looks really good, let's remove some commented code tho

**Review by ahiuchingau — APPROVED** [review]
Sweet!

**Pull request merged**